### PR TITLE
RtpsRelay doesn't have embedded webserver for meta discovery

### PR DESCRIPTION
**Pull request by jrw972**
Problem
-------

Clients of an RtpsRelay must be configured with the SPDP, SEDP, and
DATA ports of an RtpsRelay instance.  This can be done statically or
dynamically.  In the dynamic scenario, it is convenient to use
TCP/HTTP load balancing to select an instance.  The instance can
return its ports in response to an HTTP request from the client.  For
load balancing, it is useful to include a health check endpoint that
is tied to the RtpsRelay admission control.

The HTTP server to provide this functionality can be implemented
external to the RtpsRelay.  However, deployments will be simplified by
including an embedded webserver in the RtpsRelay can serve the
configuration and answer the healthcheck.

Solution
--------

Embed a webserver in the RtpsRelay that can respond to configuration
and healthcheck requests.

### CODE REVIEW

File: tools/rtpsrelay/RelayHttpMetaDiscovery.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+void RelayHttpMetaDiscovery::respondHealthcheck(std::stringstream& response) const
+{
+  GuidAddrSet::Proxy proxy(guid_addr_set_);
+
+  if (proxy.admitting()) {
+    respondStatus(response, HTTP_OK);
+  } else {
+    respondStatus(response, HTTP_SERVICE_UNAVAILABLE);
+  }
```
Comment (mitza-oci) [inline tools/rtpsrelay/RelayHttpMetaDiscovery.cpp:133]:
```suggestion
  respondStatus(response, proxy.admitting() ? HTTP_OK : HTTP_SERVICE_UNAVAILABLE);
```

File: tools/rtpsrelay/RelayHttpMetaDiscovery.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+bool RelayHttpMetaDiscovery::processRequest(ACE_SOCK_Stream& peer,
+                                            const std::string& request) const
+{
+  std::string target;
+  if (requestIsComplete(request, target)) {
+    std::stringstream response;
+    if (target == "/") {
+      respond(response);
```
Comment (mitza-oci) [inline tools/rtpsrelay/RelayHttpMetaDiscovery.cpp:69]: Why not use `relay.json` or something similar to name the resource?  It seems like that could better support extensibility.

File: tests/DCPS/RtpsRelay/Smoke/metachecker.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      ACE_ERROR((LM_ERROR, "Could not read response\n"));
+      return EXIT_FAILURE;
+    }
+    response[bytes] = 0;
+
+    ACE_DEBUG((LM_DEBUG, "Response: %C\n", response));
+
+    stream.close();
+
+    const char expected_response[]=
```
Comment (mitza-oci) [inline tests/DCPS/RtpsRelay/Smoke/metachecker.cpp:79]:
```suggestion
    const char expected_response[] =
```

File: tests/DCPS/RtpsRelay/Smoke/metachecker.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+#include <ace/Log_Msg.h>
+#include <ace/SOCK_Connector.h>
+#include <ace/SOCK_Stream.h>
+
+int ACE_TMAIN(int, ACE_TCHAR*[])
+{
+  ACE_SOCK_Connector connector;
+  ACE_SOCK_Stream stream;
+  ACE_INET_Addr remote("127.0.0.1:8081");
```
Comment (mitza-oci) [inline tests/DCPS/RtpsRelay/Smoke/metachecker.cpp:14]: configurable port?

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: tools/rtpsrelay/RtpsRelay.cpp [hunk 5]
```diff
@@ -74,6 +75,10 @@ namespace {
     }
     return ACE_INET_Addr();
   }
+
+  const unsigned short DEFAULT_HORIZONTAL = 17400;
+  const unsigned short DEFAULT_VERTICAL = 7400;
+  const unsigned short DEFAULT_META = 8080;
```
[the remark this answers is not in the document]
Comment (jrw972) [reply tools/rtpsrelay/RtpsRelay.cpp:81]: That sounds like a good reason to use 8080.

**Review by mitza-oci — APPROVED** [review]

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**